File: basic-proxy-impl/src/test/java/com/test/cases/interfaces/ConsumerTest.java
package com.test.cases.interfaces;

import org.junit.Test;

import java.util.function.Consumer;

public class ConsumerTest extends InterfaceTestBase<Consumer<String>> {

    private static final Class<?> specifiedInterface = Consumer.class;

    @Test
    public void test() throws Exception {
        @SuppressWarnings("unchecked")
        Consumer<String> standardProxy = buildStandardProxy((Class<Consumer<String>>) specifiedInterface);
        standardProxy.accept("");

        @SuppressWarnings("unchecked")
        Consumer<String> customProxy = buildCustomProxy((Class<Consumer<String>>) specifiedInterface);
        customProxy.accept("");

        verify(specifiedInterface.getMethod("accept", Object.class));
    }
}

File: basic-proxy-kotlin-impl/src/test/kotlin/com/test/cases/interfaces/ConsumerTest.kt
package com.test.cases.interfaces

import org.junit.Test

import java.util.function.Consumer

class ConsumerTest : InterfaceTestBase<Consumer<String>>() {

    private val specifiedInterface = Consumer::class.java

    @Test
    fun test() {
        val standardProxy = buildStandardProxy(specifiedInterface as Class<Consumer<String>>)
        standardProxy.accept("")

        val customProxy = buildCustomProxy(specifiedInterface as Class<Consumer<String>>)
        customProxy.accept("")

        verify(specifiedInterface.getMethod("accept", Object::class.java))
        verifyFields(standardProxy, customProxy)
    }
}
